### DISCUSSION TITLE
Bucket4j 7.6.0 with Hazelcast 5.1.3

### DISCUSSION THREAD

**Discussion by rasmusaaviste** [vladimir-bukhtoyarov/bucket4j#307] [Q&A] [answered]
Hello

I'm trying to use Hazelcast with Bucket4j and can't seem to get it working. When I try to put io.github.bucket4j.Bucket type into the map as a value, I'll get an exception "com.hazelcast.nio.serialization.HazelcastSerializationException: There is no suitable serializer for class io.github.bucket4j.local.LockFreeBucket". From the other issues I understand that I have to somehow create custom serializer for the LockFreeBucket class but don't really have an idea how to accomplish that.

Instantiating hazelcast like that: 
`    @Bean
    public HazelcastInstance hazelcastInstance() {
        return Hazelcast.newHazelcastInstance();
    }`
    
And using it like that:

`    
private final HazelcastInstance hazelcastInstance;

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws IOException {
        String bucketKey = "test-key";

        IMap<String, Bucket> map = hazelcastInstance.getMap("buckets");
        Bucket clientBucket = map.getOrDefault(bucketKey, newBucket(1));

        ConsumptionProbe probe = clientBucket.tryConsumeAndReturnRemaining(1);
        map.put(bucketKey, clientBucket);

        if (probe.isConsumed()) {
            return true;
        } else {
            response.sendError(429);

            return false;
        }
    }

    private Bucket newBucket(int limitAmount) {
        return Bucket.builder().addLimit(Bandwidth.simple(limitAmount, Duration.ofHours(1))).build();
    }`
    
Would really appreciate some guidelines how to implement custom serializer for LockFreeBucket.class.

Thanks in advance!

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @rasmusaaviste 

There is direct [integration for Hazelcast](https://bucket4j.com/8.1.0/toc.html#bucket4j-hazelcast), you do not have to reimplement it. 
Try to read documentation and [this article](https://dzone.com/articles/how-to-provide-rate-limiting-via-bucket4j-in-java).

**Reply by rasmusaaviste** [reply]
Somehow missed this article, got it successfully working. Thank you!

**Reply by kashann** [reply]
Hello,

According to the supplied documentation, `To use Bucket4j extension for Hazelcast with Hazelcast 4.x you need to add the following dependency:...`.
Is Bucket4j (`bucket4j-hazelcast`) offically compatible with Hazelcast 5.x?
If so, starting with which version?
I coudn't find any issue related to this in the [bug tracker](https://github.com/bucket4j/bucket4j/issues).

Thank you so much. I appreaciate your work 😃

**Reply by vladimir-bukhtoyarov** [reply]
> Is Bucket4j (bucket4j-hazelcast) offically compatible with Hazelcast 5.x?

Yes. Bucket4j is tested with `5.1.3` each release https://github.com/bucket4j/bucket4j/blob/master/bucket4j-parent/pom.xml#L52

> If so, starting with which version?

Since Bucket4j `7.1.0`
